coordinator/gscoordinator/kubernetes_launcher.py

```diff
@@ -744,14 +932,69 @@ def _delete_dangling_coordinator(self):
                             self._coordinator_name,
                         )
 
+    def _get_owner_reference_as_json(self):
+        owner_reference = [
+            {
+                "apiVersion": self._owner_references[0].api_version,
+                "kind": self._owner_references[0].kind,
+                "name": self._owner_references[0].name,
+                "uid": self._owner_references[0].uid,
+            }
+        ]
+        owner_reference_json = json.dumps(owner_reference)
+        return owner_reference_json
+
+    def _check_if_vineyard_deployment_exist(self):
+        try:
+            self._apps_api.read_namespaced_deployment(
+                self._vineyard_deployment, self._namespace
+            )
+        except K8SApiException:
+            logger.info(
+                f"Vineyard deployment {self._namespace}/{self._vineyard_deployment} not exist"
+            )
+            return False
+        return True
+
+    def _deploy_vineyard_deployment_if_not_exist(self):
```

Comment:
These `exists` really got me.. Why check if exists after it exists?
After some study, I think `vineyard_deployment_exists` in this context is ambiguous. in `self. vineyard_deployment_exists()` you mean the value of `self.vineyard_deployment` is not None, in ` self._check_if_vineyard_deployment_exist()`, you mean the deployment of vineyard pods is exists (This is more natural).

I think you could just use `self._vineyard_deployment_name` is not None, get rid of this method `self.vineyard_deployment_exists()`